Indexing a non-tuple struct can be used for assignment but not reading.

### Zig Version

0.11.0-dev.2868+1a455b2dd

### Steps to Reproduce and Observed Behavior

It seems like it only works when the indexing is an l-value, but not when it's an r-value.

```zig
const std = @import("std");

pub fn main() !void {
    const S = struct { x: i32, y: i32 };
    var s: S = undefined;
    s[0] = 10;
    s[1] = getNum();
    std.debug.print("{} {}\n", .{ s.x, s.y }); // <---
}

fn getNum() i32 { return 20; }
```
Output:
```
$ zig build run
10 20
```
This fails to compile:
```zig
const std = @import("std");

pub fn main() !void {
    const S = struct { x: i32, y: i32 };
    var s: S = undefined;
    s[0] = 10;
    s[1] = getNum();
    std.debug.print("{} {}\n", .{ s[0], s[1] }); // <---
}

fn getNum() i32 { return 20; }
```
Output:
```
src\main.zig:8:36: error: element access of non-indexable type 'main.main.S'
    std.debug.print("{} {}\n", .{ s[0], s[1] });
                                  ~^~~
```
Setting a value outside of the "length" of the struct gives a compile error assuming it's a tuple:
```zig
const std = @import("std");

pub fn main() !void {
    const S = struct { x: i32, y: i32 };
    @compileLog(@typeInfo(S).Struct.is_tuple);

    var s: S = undefined;
    s[2] = 100;
}
```
Output:
```
src\main.zig:8:7: error: index 2 outside tuple of length 2
    s[2] = 100;

Compile Log Output:
@as(bool, false)
```

### Expected Behavior

Most likely that any indexing of non-tuple structs will always be a compile error.

--- a/src/Sema.zig
+++ b/src/Sema.zig
@@ -24597,10 +24597,15 @@ fn elemPtr(
     const indexable_ptr_src = src; // TODO better source location
     const indexable_ptr_ty = sema.typeOf(indexable_ptr);
     const target = sema.mod.getTarget();
+
     const indexable_ty = switch (indexable_ptr_ty.zigTypeTag()) {
         .Pointer => indexable_ptr_ty.elemType(),
         else => return sema.fail(block, indexable_ptr_src, "expected pointer, found '{}'", .{indexable_ptr_ty.fmt(sema.mod)}),
     };
+    if (!indexable_ty.isIndexable()) {
+        return sema.fail(block, src, "element access of non-indexable type '{}'", .{indexable_ty.fmt(sema.mod)});
+    }
+
     switch (indexable_ty.zigTypeTag()) {
         .Array, .Vector => return sema.elemPtrArray(block, src, indexable_ptr_src, indexable_ptr, elem_index_src, elem_index, init, oob_safety),
         .Struct => {
@@ -24629,10 +24634,11 @@ fn elemPtrOneLayerOnly(
 ) CompileError!Air.Inst.Ref {
     const indexable_src = src; // TODO better source location
     const indexable_ty = sema.typeOf(indexable);
+    const target = sema.mod.getTarget();
+
     if (!indexable_ty.isIndexable()) {
         return sema.fail(block, src, "element access of non-indexable type '{}'", .{indexable_ty.fmt(sema.mod)});
     }
-    const target = sema.mod.getTarget();
 
     switch (indexable_ty.ptrSize()) {
         .Slice => return sema.elemPtrSlice(block, src, indexable_src, indexable, elem_index_src, elem_index, oob_safety),

--- a/test/cases/compile_errors/indexing_non-tuple_struct.zig
+++ b/test/cases/compile_errors/indexing_non-tuple_struct.zig
@@ -1,0 +1,11 @@
+export fn a() void {
+    const S = struct { x: i32, y: i32 };
+    var s: S = undefined;
+    s[0] = 10;
+}
+
+// error
+// backend=stage2
+// target=native
+//
+// :4:6: error: element access of non-indexable type 'tmp.a.S'

